### PR TITLE
Add support for k8s bundles

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Add support for deploying k8s bundles.
Also, other small changes so that export-bundle and diff-bundle work as well.

## QA steps

deploy a k8s bundle with relations and placement
check export-bundle
edit the bundle
check diff-bundle

### CODE REVIEW

File: cmd/juju/application/bundle.go [hunk 1]
```diff
@@ -647,6 +664,28 @@ func (h *bundleHandler) writeAddedResources(resNames2IDs map[string]string) {
 	}
 }
 
+// scaleApplication updates the number of units for an application.
+func (h *bundleHandler) scaleApplication(change *bundlechanges.ScaleChange) error {
+	// TODO: add verbose output for details
```
Comment (howbazaar) [inline cmd/juju/application/bundle.go]: Are you going to address this? What would you say in verbose that isn't in the normal output?

File: testcharms/charm-repo/bundle/kubernetes-simple/bundle.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+bundle: kubernetes
+applications:
+    gitlab:
+        charm: gitlab
+        scale: 1
+        placement: foo=bar
+    mysql:
+        charm: mysql
```
Comment (howbazaar) [inline testcharms/charm-repo/bundle/kubernetes-simple/bundle.yaml]: Shouldn't this be mariadb ;-)

File: cmd/juju/application/bundle.go (hunk 1 above)
Comment (wallyworld) [reply cmd/juju/application/bundle.go]: cut and paste error

File: testcharms/charm-repo/bundle/kubernetes-simple/bundle.yaml (hunk 2 above)
Comment (wallyworld) [reply testcharms/charm-repo/bundle/kubernetes-simple/bundle.yaml]: there needs to be a corresponding local charm. i've renamed the mysql one and updated usages in tests

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**